### PR TITLE
[sc-29547] Deprecate BQ lineage crawler and a lot of refactor

### PR DESCRIPTION
**Pull request by usefulalgorithm**
### 🤔 Why?

We want to deprecate BQ lineage crawler.

### 🤓 What?

- Removed implementation in BQ lineage crawler, and move the logic into main BQ crawler.
- Moved a bunch of code in main BQ crawler into separate files.

### 🧪 Tested?

- Run crawler against our stage configuration.
- Updated unit tests.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/bigquery/config.py [hunk 1]
```diff
@@ -69,6 +69,24 @@ class BigQueryQueryLogConfig:
...
+@dataclass(config=ConnectorConfig)
+class BigQueryLineageConfig:
+    # Whether to enable parsing view query to find upstream of the view, default True
+    enable_view_lineage: bool = True
+
+    # Whether to enable parsing audit log to find table lineage information, default True
+    enable_lineage_from_log: bool = True
+
+    # Number of days back in the query log to process
+    lookback_days: int = 7
```
Comment (mars-lan) [inline metaphor/bigquery/config.py:81]:
```suggestion
    lookback_days: int = 1
```

**Review by mars-lan — COMMENTED** [review]
Does this include the rate limit change (https://github.com/MetaphorData/connectors/pull/1032)?

**Comment by usefulalgorithm** [comment]
It does

**Review by mars-lan — APPROVED** [review]
Thanks for refactoring!

**Pull request merged**